src/app/products/loan-products/create-loan-product/create-loan-product.component.ts

```diff
@@ -139,12 +174,16 @@ export class CreateLoanProductComponent implements OnInit {
       loanProduct['dueDaysForRepaymentEvent'] = null;
       loanProduct['overDueDaysForRepaymentEvent'] = null;
     }
+    loanProduct['supportedInterestRefundTypes'] = this.mapStringEnumOptionToIdList(loanProduct['supportedInterestRefundTypes']);
     delete loanProduct['useDueForRepaymentsConfigurations'];
 
     this.productsService.createLoanProduct(loanProduct)
       .subscribe((response: any) => {
-        this.router.navigate(['../', response.resourceId], { relativeTo: this.route });
+        this.router.navigate(['../', response.resourceId], {relativeTo: this.route});
       });
   }
 
+  mapStringEnumOptionToIdList(incomingValues: StringEnumOptionData[]): string[] {
```

Comment:
Use the `OptionData` already defined in 'app/shared/models/option-data.model'

Reply:
Its not the same. OptionData id is number. StringEnumOptionData id is string